packages/shared/src/components/squads/settings/SquadPrivacySection.tsx

```diff
@@ -0,0 +1,132 @@
+import {
+  SourceCategoryData,
+  SOURCE_CATEGORIES_QUERY,
+} from '../../../graphql/sources';
+import {
+  generateQueryKey,
+  RequestKey,
+  flattenInfiniteQuery,
+} from '../../../lib/query';
+import { Dropdown } from '../../fields/Dropdown';
+import { Radio } from '../../fields/Radio';
+import { StatusDescription } from './common';
+import { SquadSettingsSection } from './SquadSettingsSection';
+
+interface CategoryDropdownProps {
+  initialCategory: string;
+  isPublic: boolean;
+  categoryHint?: string;
+  onCategoryChange?: (category: string) => void;
+}
+
+const classes = {
+  wrapper: 'w-full',
+  content: 'w-fit',
+};
+
+export enum PrivacyOption {
+  Private = 'private',
+  Public = 'public',
+}
+
+export function SquadPrivacySection({
+  initialCategory,
+  isPublic = true,
+  categoryHint,
+  onCategoryChange,
+}: CategoryDropdownProps): ReactElement {
+  const [privacy, setPrivacy] = useState(
+    isPublic ? PrivacyOption.Public : PrivacyOption.Private,
+  );
+  const isPrivate = privacy === PrivacyOption.Private;
+  const [selected, setSelected] = useState(-1);
+  const { data } = useInfiniteQuery<SourceCategoryData>(
+    generateQueryKey(RequestKey.Source, null, 'categories'),
+    ({ pageParam }) => gqlClient.request(SOURCE_CATEGORIES_QUERY, pageParam),
+  );
+  const list = useMemo(() => flattenInfiniteQuery(data, 'categories'), [data]);
+
+  useEffect(() => {
+    if (!list?.length) {
+      return;
+    }
+
+    const index = list.findIndex(({ id }) => id === initialCategory);
+
+    if (index === -1) {
+      return;
+    }
+
+    setSelected(index);
+  }, [initialCategory, list]);
+
+  const privacyOptions = useMemo(() => {
+    return [
+      {
+        label: 'Public',
+        value: PrivacyOption.Public,
+        className: classes,
+        afterElement: (
+          <div className="ml-9 flex flex-col gap-4">
+            <StatusDescription>
+              Squad is searchable, publicly listed in our squad directory, and
+              open for everyone to join. It’s ideal for building a community or
+              for content creators.
+            </StatusDescription>
+            {list && (
+              <Dropdown
+                placeholder="Select category"
+                className={{
+                  container: 'w-56',
+                  button: categoryHint && 'border border-status-error',
+                }}
+                options={list.map(({ title }) => title)}
+                selectedIndex={selected}
+                onChange={(_, index) => {
+                  setSelected(index);
+                  onCategoryChange?.(list[index].id);
+                }}
+                disabled={isPrivate}
+              />
+            )}
+            {list && !isPrivate && (
+              <input
+                type="hidden"
+                id="categoryId"
+                name="categoryId"
+                value={list[selected]?.id}
+              />
+            )}
+          </div>
```

Comment:
Let's move this into a component, even if we want to keep it in the same file

Reply:
I tried to, but there's a lot of moving objects on this section, and decided to leave it here.

Reply:
It could be probably within the component no? that way u don't need to pass anything just separating the code

Reply:
Yes, that's what I hoped when I tried it at first, but most values here are used within this whole form.

Reply:
I am doing some refactoring on the form as a whole, I will check again if something can still be extracted easily.

Reply:
Yeah cool, no worries if it brings too much trouble

Reply:
My mistake. I overlooked it, I thought the comment was referring to the name/handle/description being extracted too. This was about the dropdown. You're right. We should be able to move this one.